src/bokeh/models/coordinates.py

```diff
@@ -68,10 +78,57 @@ def __init__(self, *args, **kwargs) -> None:
     The vertical range to map y-coordinates in the target coordinate space.
     """)
 
+class Node(Model):
+    """
+    Represents a symbolic coordinate (by name).
+
+    .. note::
+        This model is experimental and may change at any point.
+    """
+
+    # explicit __init__ to support Init signatures
+    def __init__(self, *args, **kwargs) -> None:
+        super().__init__(*args, **kwargs)
+
+    target = Required(Either(Instance(Model), Enum("canvas", "plot", "frame", "parent")), help="""
+    The provider of coordinates for this node.
+
+    This can be either a concrete model that can provide its coordinates (e.g.
+    a renderer, a frame or a canvas) or an implicit target defined by the
+    enum, which is resolved as the nearest parent of the given type. If the
+    provider cannot be determined or it isn't able to provide coordinates,
+    then the node resolved to an invalid coordinate (with x and y components
+    being ``NaN``).
+    """)
+
+    term = Required(String(), help="""
```

Comment:
```suggestion
    term = Required(String, help="""
```
would be more consistent with existing usage elsewhere

Reply:
Also "term" doesn't really say anything to me, at all. If anything it suggests something more like a time duration ("semester term") What about just "symbol" instead?

Reply:
We have, for example, `:term:` directive in documentation. I took this name from logic programming. However, "symbol" does indeed seem more meaningful in this context.